### PR TITLE
fix: add variant to manifest entry 

### PR DESCRIPTION
**Pull request by thesayyn**
part of #362

### CODE REVIEW

File: oci/tests/BUILD.bazel [hunk 1]
```diff
@@ -16,7 +16,12 @@ IMAGES_TO_TEST = {
     genrule(
         name = "pull_{}".format(repo_name),
         outs = [repo_name],
-        cmd = "$(CRANE_BIN) pull {reference} $@ --format=oci --platform={platform}".format(
+        cmd = """
+$(CRANE_BIN) pull {reference} $@ --format=oci --platform={platform}
+cat "$@/index.json" | $(JQ_BIN) --sort-keys -j '.' > "$@/index-tmp.json"
```
Comment (alexeagle) [inline oci/tests/BUILD.bazel:21]: needs some comments about doing this to make it match the json.encode_indent output that starlark is making

**Review by alexeagle — COMMENTED** [review]
Could you help me understand why it's an improvement? It feels approx the same performance and complexity overall

**Comment by thesayyn** [comment]
> Could you help me understand why it's an improvement? It feels approx the same performance and complexity overall

the `json.encode_ident` does not provide any performance improvement over the templated json, it is purely nice to have to allow more properties to be added to the platform sub-object conditionally without templating tricks. https://github.com/opencontainers/image-spec/blob/main/image-index.md#image-index-property-descriptions
 I can revert that. 
 
That said the use of write_file is unnecessary here and adds some performance overhead. eg: spawning some actions to write a file which is pure overhead given that we already write the same content to the BUILD file.

**Comment by thesayyn** [comment]
actually, now that I think about it more, I will keep that part the same just make it more ergonomic. eg: instead of two templates, just use one.

File: oci/tests/BUILD.bazel (hunk 1 above)
Comment (thesayyn) [reply oci/tests/BUILD.bazel]: not relevant anymore.

**Comment by thesayyn** [comment]
@alexeagle i took back the templated strings back and made an util function that can add `variant` as well. now we match `crane pull` 100% of the cases.  

now we have the ability to write a starlark unit test for the util function.  see #138

**Review by alexeagle — APPROVED** [review]

**Pull request merged**